src/main/java/fi/vrk/xroad/monitor/actor/Supervisor.java

```diff
@@ -0,0 +1,98 @@
+package fi.vrk.xroad.monitor.actor;
+
+import akka.actor.*;
+import akka.dispatch.Futures;
+import akka.japi.pf.DeciderBuilder;
+import akka.pattern.Patterns;
+import akka.routing.SmallestMailboxPool;
+import akka.util.Timeout;
+import fi.vrk.xroad.monitor.extensions.SpringExtension;
+import fi.vrk.xroad.monitor.parser.SecurityServerInfo;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+import scala.concurrent.Await;
+import scala.concurrent.Future;
+import scala.concurrent.duration.Duration;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static akka.actor.SupervisorStrategy.*;
+import static fi.vrk.xroad.monitor.util.MonitorCollectorConstants.SUPERVISOR_MONITOR_DATA_ACTOR_POOL_SIZE;
+
+/**
+ * Supervisor for actors
+ */
+@Component
+@Scope("prototype")
+@Slf4j
+public class Supervisor extends AbstractActor {
+
+  private ActorRef monitorDataRequestPoolRouter;
+  private String workerActorName;
+
+  @Autowired
+  private SpringExtension springExtension;
+
+  public Supervisor(ActorRef monitorDataRequestPoolRouter, String workerActorName) {
+    this.monitorDataRequestPoolRouter = monitorDataRequestPoolRouter;
+    this.workerActorName = workerActorName;
+  }
+
+  @Override
+  public void preStart() throws Exception {
+    super.preStart();
+    log.info("preStart");
+  }
+
+  @Override
+  public Receive createReceive() {
+    log.info("createReceive");
+    return receiveBuilder()
+        .match(StartCollectingMonitorDataCommand.class, this::handleMonitorDataRequest)
+        .build();
```

Comment:
What happens to messages that are not StartCollectingMonitorDataCommands (maybe sent due to error)? If receiveBuilder does not have any internal magic, would be good to:

.matchAny(this::unhandled)

as in https://opencredo.com/improved-akka-java-8/ to not silently fail. 

Also register actor like https://github.com/vrk-kpa/X-Road/blob/PVAYLADEV-746-monitor-certificate-expiration/src/monitor/src/main/java/ee/ria/xroad/monitor/UnhandledListenerActor.java OR configure Akka logging so that those errors become visible.